src/signTx.c

```diff
@@ -1294,10 +1183,87 @@ static void signTx_handleMetadataAPDU(uint8_t p2, uint8_t* wireDataBuffer, size_
 	signTx_handleMetadata_ui_runStep();
 }
 
+// ============================== VALIDITY INTERVAL START ==============================
+
+enum {
+	HANDLE_VALIDITY_INTERVAL_STEP_DISPLAY = 900,
+	HANDLE_VALIDITY_INTERVAL_STEP_RESPOND,
+	HANDLE_VALIDITY_INTERVAL_STEP_INVALID,
+};
+
+static void signTx_handleValidityInterval_ui_runStep()
+{
+	TRACE("UI step %d", ctx->ui_step);
+	ui_callback_fn_t* this_fn = signTx_handleValidityInterval_ui_runStep;
+
+	UI_STEP_BEGIN(ctx->ui_step);
+
+	UI_STEP(HANDLE_VALIDITY_INTERVAL_STEP_DISPLAY) {
+		ui_displayUint64Screen(
+		        "Validity interval start",
+		        ctx->stageData.validityIntervalStart,
+		        this_fn
+		);
+	}
+	UI_STEP(HANDLE_VALIDITY_INTERVAL_STEP_RESPOND) {
+		respondSuccessEmptyMsg();
+		advanceStage();
+	}
+	UI_STEP_END(HANDLE_VALIDITY_INTERVAL_STEP_INVALID);
+}
+
+static void signTx_handleValidityIntervalAPDU(uint8_t p2, uint8_t* wireDataBuffer, size_t wireDataSize)
```

Comment:
I'd rather name it `signTx_handleValidityIntervalStartAPDU` given that `validityInterval` alone can mislead the reader into thinking that both validityIntervalStart and TTL are being handled here